src/core/qgsmapthemecollection.h

```diff
@@ -150,12 +151,23 @@ class CORE_EXPORT QgsMapThemeCollection : public QObject
          */
         bool hasExpandedStateInfo() const { return mHasExpandedStateInfo; }
 
+        /**
+         * Returns whether information about checked/unchecked state of groups has been recorded
+         * and thus whether checkedGroupNodes() is valid.
+         */
+        bool hasCheckedStateInfo() const { return mHasCheckedStateInfo; }
```

Comment:
See below -- I'd also suggest hiding this from the sip python bindings, as it's just an internal detail rather then something we want to expose